hudi-utilities/src/main/java/org/apache/hudi/utilities/transform/ChainedTransformer.java

```diff
@@ -59,11 +65,15 @@ public ChainedTransformer(List<Transformer> transformersList) {
    * Creates a chained transformer using the input transformer class names. Refer {@link HoodieDeltaStreamer.Config#transformerClassNames}
    * for more information on how the transformers can be configured.
    *
-   * @param configuredTransformers List of configured transformer class names.
-   * @param ignore Added for avoiding two methods with same erasure. Ignored.
+   * @param sourceSchemaOpt                   Source Schema
+   * @param configuredTransformers            List of configured transformer class names.
+   * @param enableSchemaValidation            If true, schema is validated for the transformed data against expected schema.
+   *                                          Expected schema is provided by {@link Transformer#transformedSchema}
    */
-  public ChainedTransformer(List<String> configuredTransformers, int... ignore) {
+  public ChainedTransformer(List<String> configuredTransformers, Option<Schema> sourceSchemaOpt, boolean enableSchemaValidation) {
     this.transformers = new ArrayList<>(configuredTransformers.size());
+    this.enableSchemaValidation = enableSchemaValidation;
+    this.sourceSchemaOpt = sourceSchemaOpt;
```

Comment:
do we need any validation as below
when enableSchemaValidation is true, sourceSchemaOpt is present ?

Reply:
This is handled in the schema validation check `validateAndGetTransformedSchema`. The first incomingSchema is the sourceSchemaOpt.